Add `CHANGELOG.md`

mokagio opened the pull request:
Adds `CHANGELOG.md`. Based on https://github.com/wordpress-mobile/WordPressKit-iOS/pull/545.

It also occurred to me that we ought to let folks know about the file, so I added a note about it the (created for the occasion) PR template.

mokagio committed:
Add `CHANGELOG.md`

mokagio committed:
Add changelog entry about changelog itself

mokagio committed:
Add `PULL_REQUEST_TEMPLATE.md` with reminder about `CHANGELOG.md`

AliSoftware commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,4 @@
+
+---
+
+- [ ] I have considered if this change warrants release notes and have added them to `CHANGELOG.md` if necessary.
```
👍

AliSoftware replied:
```suggestion
- [ ] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary.
```

mokagio committed:
Improve `CHANGELOG.md` instructions

Co-authored-by: Olivier Halligon <olivier.halligon@automattic.com>

mokagio committed:
Update bash-cache Buildkite plugin to address Bundler failure

See
https://buildkite.com/automattic/wordpress-ui-ios/builds/68#0183efb4-1c70-4c68-b5ca-f4c5002a9dc6
